packages/plugins/plugin-idfa/src/IdfaPlugin.tsx

```diff
@@ -3,36 +3,64 @@ import {
   PluginType,
   SegmentClient,
 } from '@segment/analytics-react-native';
-import { IdfaEvents } from './IdfaEvents';
 import type { IdfaData } from './types';
 import { AnalyticsReactNativePluginIdfa } from './AnalyticsReactNativePluginIdfa';
 
 const { getTrackingAuthorizationStatus } = AnalyticsReactNativePluginIdfa;
 
 export class IdfaPlugin extends Plugin {
   type = PluginType.enrichment;
+  private isEnabled?: boolean = false;
+  private isDisabled?: boolean;
+
+  constructor(enabled?: boolean) {
+    super();
+    if (enabled === false) {
+      this.isDisabled = true;
+    }
+
+    if (this.isDisabled !== true) {
+      this.getTrackingStatus();
+    }
+  }
 
   configure(analytics: SegmentClient) {
     this.analytics = analytics;
 
-    this.getTrackingStatus();
-
-    // subscribe to IDFAQuery event
-    // emitted when we prompt a user for permission
-    IdfaEvents.addListener('IDFAQuery', (res) => {
+    // since configure can be called multiple times potentially
+    // this accounts for both enabling and disabling plugin
+    if (this.isEnabled === false && this.isDisabled !== true) {
       this.getTrackingStatus();
-      this.analytics?.track('IDFAQuery', res);
-    });
+    }
```

Comment:
We shouldn't ask for permissions on `configure`. This is only for Destination plugins that get new settings, but these settings are only set programatically, furthermore these are private so the user cannot change them afterwards.